src/server/project.ts

```diff
@@ -2215,6 +2215,20 @@ export abstract class Project implements LanguageServiceHost, ModuleResolutionHo
         return this.noDtsResolutionProject;
     }
 
+    /** @internal */
+    updateTargetFile(rootFile: string, targetFileText: string, pastedText: string): { updatedFile: SourceFile | undefined; updatedProgram: Program | undefined; originalProgram: ts.Program | undefined; } {
+        const originalProgram = this.program;
+        this.getScriptInfo(rootFile)?.editContent(0, targetFileText.length, pastedText);
+        this.updateGraph();
+        return { updatedFile: (this.program?.getSourceFile(rootFile)), updatedProgram: this.program, originalProgram };
+    }
+
+    /** @internal */
+    revertUpdatedFile(rootFile: string, updatedText: string, originalText: string) {
+        this.getScriptInfo(rootFile)?.editContent(0, updatedText.length, originalText);
+        this.updateGraph();

```

Comment:
i dont think you should do updateGraph here.,..  (that should take place whenever its needed instead)

Reply:
If I skip updateGraph from here then the file does not get back to its original form, with the original text. So to ensure that happens, I added updateGraph here.

Reply:
The file will be updated at next update.